template/src/components/Toast.tsx

```diff
@@ -1,202 +1,175 @@
-            useNativeDriver: true,
-          }),
-          Animated.timing(this.offset, {
-            toValue: HEIGHT * -1,
-            duration: 300,
-            useNativeDriver: true,
-          }),
-        ]),
-      ])
-      this.animated.start()
-    })
-  }
-
-  dismiss = () => {
-    this.animated?.stop()
-    Animated.parallel([
-      Animated.parallel([
-        Animated.timing(this.opacity, {
-          toValue: 0,
-          duration: 300,
+        Animated.timing(animation, {
+          toValue: 1,
+          duration: args.option?.duration || DEFAULT_DURATION,
           useNativeDriver: true,
         }),
-        Animated.timing(this.offset, {
-          toValue: HEIGHT * -1,
-          duration: 300,
+        Animated.delay(args.option?.delay || DEFAULT_DELAY),
+        // Fade Out
+        Animated.timing(animation, {
+          toValue: 0,
+          duration: args.option?.duration || DEFAULT_DURATION,
           useNativeDriver: true,
         }),
-      ]),
-    ]).start()
+      ])
+      animationRef.current.start(() => {
+        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
+      })
+    }, 100)
   }
 
-  messageColor = (): string => {
-    const {type} = this.state
-
-    if (type === EToastType.SUCCESS) {
-      return colors.success
-    }
-    if (type === EToastType.INFO) {
-      return colors.info
-    }
-    return colors.error
+  const dismiss = () => {
+    animationRef.current?.stop()
+    setState(initState)
+    Animated.timing(animation, {
+      toValue: 0,
+      duration: DEFAULT_DURATION,
+      useNativeDriver: true,
+    }).start()
   }
 
-  render() {
-    const {message, type} = this.state
-    return (
-      <Animated.View
+  useEffect(() => {
+    Emitter.on(TOAST_MESSAGE, displayMessage)
+    return () => {
+      Emitter.rm(TOAST_MESSAGE)
+    }
+  }, [])
+
+  return (
+    <Animated.View
+      style={[
+        styles.container,
+        {
+          transform: [{translateY: offset}],
+          opacity: opacity,
+          paddingBottom: insets.bottom,
+        },
+      ]}>
+      <View
         style={[
-          styles.container,
+          styles.messageContainer,
           {
-            transform: [{translateY: this.offset}],
-            opacity: this.opacity,
-            backgroundColor: this.messageColor(),
+            backgroundColor: messageContent[state.type].color,
           },
         ]}>
-        <TouchableOpacity
-          activeOpacity={1}
-          style={styles.messageContainer(this.messageColor())}
-          onPress={() => {
-            this.dismiss()
-          }}>
-          <Text style={styles.textStyle}>
-            {message}
-          </Text>
+        <View style={styles.textContent}>
+          <Text style={styles.titleStyle}>{state.message}</Text>
+          {!!state.subMessage && <Text style={styles.textStyle}>{state.subMessage}</Text>}
+        </View>
+        <TouchableOpacity onPress={dismiss} hitSlop={hitSlop}>
```

Comment:
Should create a rightComponent here and set this to default

Reply:
Basically this Toast is not a component, we call it by `showToast()` action, so we don't need to customize it too much
If you need to customize it base on design, you can customize it after you have init a new project

Reply:
As last RN team meeting discussion, we need to implement the component more flexible as much as we can

Reply:
As I said above, this is not a component but something like alert of an error, so the customization for this should be base on each project. We should not customize this too much here
Rather do it after we init the project